node/node.go

```diff
@@ -374,6 +383,60 @@ func (n *Node) initNetworking(primaryNetVdrs validators.Set) error {
 	return err
 }
 
+type NodeRuntimeState struct {
+	// The process id of the node
+	PID int
+	// URI to access the node API
+	// Format: [https|http]://[host]:[port]
+	URI string
+	// Address other nodes can use for bootstrapping
+	// Format: [host]:[port]
+	BootstrapAddress string
+}
+
+// Write runtime state to the configured path. Supports the use of
+// dynamically chosen network ports with local network orchestration.
```

Comment:
If we wrote just the pid to a predetermined path could we conclude the ports that process is listening on from that?

Reply:
In the same way `ss` or `netstat` would reading `/proc/`. Have not tried but I think it could be possible with `procfs` [1]  to parse `/proc/${PID}/net/tcp`.

```
ss -tlp
```

[1] https://github.com/prometheus/procfs